types/mempool/sender_nonce_property_test.go

```diff
@@ -0,0 +1,117 @@
+package mempool_test
+
+import (
+	"math/rand"
+	"sort"
+
+	"pgregory.net/rapid"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	mempool "github.com/cosmos/cosmos-sdk/types/mempool"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+var (
+	_ sdk.Tx                  = (*testTx)(nil)
+	_ signing.SigVerifiableTx = (*testTx)(nil)
+	_ cryptotypes.PubKey      = (*testPubKey)(nil)
+)
+
+// Property Based Testing
+// Split the senders tx in independent slices and then test the following properties in each slice
+// same elements input output and overwrite the nonce duplicates
```

Comment:
> // same elements input output and overwrite the nonce duplicates

what does "same elements input output" mean?

Reply:
I meant this same elements input on the mempool should be in the output except for sender nonce duplicates, which are overwritten by the later duplicate entries.